src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1408,15 +1408,26 @@ public Key(File file, BasicFileAttributes attrs, ZipCoder zc) {
 
             public int hashCode() {
                 long t = utf8 ? 0 : Long.MAX_VALUE;
+                /*
+                 * The lastModifiedTime attribute is used to detect cases where
+                 * an existing file is updated. The other hash code properties
+                 * by themselves won't detect such a change.
+                 */
```

Comment:
I see you've added a comment to the checking of the last modified time but I think the comment is a bit confusing as there isn't any detection of a changed zip file in the hashCode/equals methods.  I think you want to say that the lastModifiedTime is included in the hash code to allow for cases where the same file is opened more than once and where it has been modified in the mean-time.